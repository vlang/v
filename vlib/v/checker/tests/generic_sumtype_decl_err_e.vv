type Fn[T] = fn (string) T

type Fnn = fn (string) string

type Parser[T] = Fn[T] | Fnn

fn f(x string) string {
	return '>> f ${x}'
}

fn main() {
	_ := Parser(f)
}
