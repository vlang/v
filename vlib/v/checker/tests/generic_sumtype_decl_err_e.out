vlib/v/checker/tests/generic_sumtype_decl_err_e.vv:12:7: error: generic sumtype `Parser` must specify type parameter, e.g. Parser[int]
   10 | 
   11 | fn main() {
   12 |     _ := Parser(f)
      |          ~~~~~~~~~
   13 | }
